src/control/tools/EditSelection.h

```diff
@@ -32,7 +34,7 @@ class UndoAction;
 class EditSelectionContents;
 class DeleteUndoAction;
 
-class EditSelection: public ElementContainer, public Serializeable {
+class EditSelection: public ElementContainer, public Serializeable, public SnapToGridInputHandler {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.